zetaclient/context/app_context.go

```diff
@@ -0,0 +1,242 @@
+// Package context provides global app context for ZetaClient
+package context
+
+import (
+	"sort"
+	"sync"
+
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/rs/zerolog"
+
+	"github.com/zeta-chain/zetacore/pkg/chains"
+	lightclienttypes "github.com/zeta-chain/zetacore/x/lightclient/types"
+	observertypes "github.com/zeta-chain/zetacore/x/observer/types"
+	"github.com/zeta-chain/zetacore/zetaclient/config"
+)
+
+// AppContext contains zetaclient application context
+// these are initialized and updated at runtime periodically
+type AppContext struct {
+	config           config.Config
+	keygen           observertypes.Keygen
+	currentTssPubkey string
+	chainsEnabled    []chains.Chain
+	chainParamMap    map[int64]*observertypes.ChainParams
+	btcNetParams     *chaincfg.Params
+	crosschainFlags  observertypes.CrosschainFlags
+
+	// additionalChains is a list of additional static chain information to use when searching from chain IDs
+	// it is stored in the protocol to dynamically support new chains without doing an upgrade
+	additionalChain []chains.Chain
+
+	// blockHeaderEnabledChains is used to store the list of chains that have block header verification enabled
+	// All chains in this list will have Enabled flag set to true
+	blockHeaderEnabledChains []lightclienttypes.HeaderSupportedChain
+
+	// mu is to protect the app context from concurrent access
+	mu sync.RWMutex
+}
+
+// New creates empty app context with given config
+func New(cfg config.Config) *AppContext {
```

Comment:
For the 1st comment. I had hard time to resolve tons of conflicts. In case I missed something, can you point out what was overwritten? I saw the `New` method look like similar to this new version.
![image](https://github.com/zeta-chain/node/assets/34498985/e6e6d59c-8131-41a9-9d8d-9e99c250f00f)

Reply:
Can this pause merging this PR unless we merge https://github.com/zeta-chain/node/pull/2428? I'll help you fix merge conflicts. cc @lumtis